[feat] engine: implementation of geizhals.de

Bnyro opened the pull request:
## What does this PR do?
- add support for geizhals.de

## Why is this change important?
- because geizhals.de can basically "replace" all other shopping search engines in Germany

## How to test this PR locally?
- `!geizhals sort:asc monitor`
- `!geizhals tastatur`

Bnyro committed:
[feat] engine: implementation of geizhals.de

return42 commented on searx/engines/geizhals.py:
```diff
@@ -0,0 +1,86 @@
+base_url = "https://geizhals.de"
+sort_order = 'relevance'
+
+SORT_RE = re.compile("sort:(\w+)")
```
String constant might be missing an `r` prefix. (anomalous-backslash-in-string) --> `r"sort:(\w+)`

return42 commented on searx/engines/geizhals.py:
```diff
@@ -0,0 +1,87 @@
+sort_order = 'relevance'
+
+SORT_RE = re.compile(r"sort:(\w+)")
+sort_order_map = {
+    'relevance': None,
+    'price': 'p',
+    'asc': 'p',
+    'desc': '-p',
+}
```
Whats the difference of `price` and `asc` in a search term?

Bnyro replied:
It's the same, it's just for convenience so thay users can use both

return42 committed:
[fix] products template: don't quote html tags in result.content

The result.content field is *safe* HTML, tags to highlight search terms are
intended.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 reviewed (approved):
Thanks a lot for adding geizhals to SearXNG / I often use this page while I'm on shopping :-)

I slightly modified your commit to add the doc strings to the online documentation and BTW: I fixed an issue in the products.html in c3cc660ea.
